src/ImageDeleteModal.jsx

```diff
@@ -17,115 +17,96 @@ function sortTags(a, b) {
+        acc[item] = (i === 0);
+        return acc;
+    }, {}));
+
+    const checkedTags = Object.keys(tags).sort(sortTags)
+            .filter(x => tags[x]);
+
+    const onValueChanged = (item, value) => {
+        setTags(prevState => ({
+            ...prevState,
+            [item]: value,
+        }));
+    };
+
+    const pickAll = () => {
+        setTags(prevState => Object.keys(prevState).reduce((acc, item, i) => {
+            acc[item] = true;
+            return acc;
+        }, {}));
+    };
+
+    const handleRemoveImage = (tags, all) => {
         const handleForceRemoveImage = () => {
             Dialogs.close();
-            return client.delImage(image.isSystem, image.Id, true)
+            return client.delImage(imageWillDelete.isSystem, imageWillDelete.Id, true)
                     .catch(ex => {
-                        const error = cockpit.format(_("Failed to force remove image $0"), image.RepoTags[0]);
-                        this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                        const error = cockpit.format(_("Failed to force remove image $0"), imageWillDelete.RepoTags[0]);
+                        onAddNotification({ type: 'danger', error, errorDetail: ex.message });
                         throw ex;
                     });
         };
 
         Dialogs.close();
         if (all)
-            client.delImage(image.isSystem, image.Id, false)
+            client.delImage(imageWillDelete.isSystem, imageWillDelete.Id, false)
                     .catch(ex => {
-                        Dialogs.show(<ForceRemoveModal name={image.RepoTags[0]}
+                        Dialogs.show(<ForceRemoveModal name={imageWillDelete.RepoTags[0]}
                                                        handleForceRemove={handleForceRemoveImage}
                                                        reason={ex.message} />);
                     });
         else {
             // Call another untag once previous one resolved. Calling all at once can result in undefined behavior
             const tag = tags.shift();
             const i = tag.lastIndexOf(":");
-            client.untagImage(image.isSystem, image.Id, tag.substring(0, i), tag.substring(i + 1, tag.length))
+            client.untagImage(imageWillDelete.isSystem, imageWillDelete.Id, tag.substring(0, i), tag.substring(i + 1, tag.length))
                     .then(() => {
                         if (tags.length > 0)
-                            this.handleRemoveImage(tags, all);
+                            handleRemoveImage(tags, all);
                     })
                     .catch(ex => {
                         const error = cockpit.format(_("Failed to remove image $0"), tag);
-                        this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                        onAddNotification({ type: 'danger', error, errorDetail: ex.message });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-163743-c39fb479-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)